examples/chef/common/chef-resource-monitoring-delegates.cpp

```diff
@@ -0,0 +1,156 @@
+
+const chip::BitMask<ResourceMonitoring::Feature>
+    gHepaFilterFeatureMap(static_cast<uint32_t>(ResourceMonitoring::Feature::kCondition) |
+                          static_cast<uint32_t>(ResourceMonitoring::Feature::kWarning) |
+                          static_cast<uint32_t>(ResourceMonitoring::Feature::kReplacementProductList));
+const chip::BitMask<ResourceMonitoring::Feature>
+    gActivatedCarbonFeatureMap(static_cast<uint32_t>(ResourceMonitoring::Feature::kCondition) |
+                               static_cast<uint32_t>(ResourceMonitoring::Feature::kWarning) |
+                               static_cast<uint32_t>(ResourceMonitoring::Feature::kReplacementProductList));
+
+static std::unique_ptr<ActivatedCarbonFilterMonitoringDelegate> gActivatedCarbonFilterDelegate = nullptr;
+static std::unique_ptr<ResourceMonitoring::Instance> gActivatedCarbonFilterInstance            = nullptr;
+
+static std::unique_ptr<HepaFilterMonitoringDelegate> gHepaFilterDelegate = nullptr;
+static std::unique_ptr<ResourceMonitoring::Instance> gHepaFilterInstance = nullptr;
+
+static ImmutableReplacementProductListManager sReplacementProductListManager;
+
+//-- Activated Carbon Filter Monitoring delegate methods
+CHIP_ERROR ActivatedCarbonFilterMonitoringDelegate::Init()
+{
+    ChipLogDetail(Zcl, "ActivatedCarbonFilterMonitoringDelegate::Init()");
+    GetInstance()->SetReplacementProductListManagerInstance(&sReplacementProductListManager);
+    return CHIP_NO_ERROR;
+}
+
+Status ActivatedCarbonFilterMonitoringDelegate::PreResetCondition()
+{
+    ChipLogDetail(Zcl, "ActivatedCarbonFilterMonitoringDelegate::PreResetCondition()");
+    return Status::Success;
+}
+
+Status ActivatedCarbonFilterMonitoringDelegate::PostResetCondition()
+{
+    ChipLogDetail(Zcl, "ActivatedCarbonFilterMonitoringDelegate::PostResetCondition()");
+    return Status::Success;
+}
+
+void ActivatedCarbonFilterMonitoring::Shutdown()
+{
+    gActivatedCarbonFilterInstance = nullptr;
+    gActivatedCarbonFilterDelegate = nullptr;
+}
+
+//-- Hepa Filter Monitoring delegate methods
+CHIP_ERROR HepaFilterMonitoringDelegate::Init()
+{
+    ChipLogDetail(Zcl, "HepaFilterMonitoringDelegate::Init()");
+    GetInstance()->SetReplacementProductListManagerInstance(&sReplacementProductListManager);
+    return CHIP_NO_ERROR;
+}
+
+Status HepaFilterMonitoringDelegate::PreResetCondition()
+{
+    ChipLogDetail(Zcl, "HepaFilterMonitoringDelegate::PreResetCondition()");
+    return Status::Success;
+}
+
+Status HepaFilterMonitoringDelegate::PostResetCondition()
+{
+    ChipLogDetail(Zcl, "HepaFilterMonitoringDelegate::PostResetCondition()");
+    return Status::Success;
+}
+
+void HepaFilterMonitoring::Shutdown()
+{
+    gHepaFilterInstance = nullptr;
+    gHepaFilterDelegate = nullptr;
+}
+
+void emberAfActivatedCarbonFilterMonitoringClusterInitCallback(chip::EndpointId endpoint)
+{
+    VerifyOrDie(gActivatedCarbonFilterInstance == nullptr && gActivatedCarbonFilterDelegate == nullptr);
```

Comment:
I think you can use it directly without nullptr: https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool is a thing

Reply:
Fixed done